openverse_catalog/dags/common/pr_review_reminders.py

```diff
@@ -0,0 +1,161 @@
+import logging
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Optional
+
+from common.github import GitHubAPI
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class Urgency:
+    label: str
+    days: int
+
+
+@dataclass
+class ReviewDelta:
+    urgency: Urgency
+    days: int
+
+
+def pr_urgency(pr) -> Urgency:
+    priority_label = [
+        label["name"] for label in pr["labels"] if "priority" in label["name"].lower()
+    ][0]
+    if "critical" in priority_label:
+        return Urgency("critical", 1)
+    elif "high" in priority_label:
+        return Urgency("high", 2)
+    elif "medium" in priority_label:
+        return Urgency("medium", 4)
+    elif "low" in priority_label:
+        return Urgency("low", 5)
+
+
+def days_without_weekends(today: datetime, delta: timedelta) -> int:
+    days_in_previous_week = abs(today.weekday() - delta.days)
+    if days_in_previous_week > 0:
+        if days_in_previous_week < 2:
+            return 0
+        return abs(delta.days - max((days_in_previous_week // 7) * 2, 2))
+
+    return delta.days
+
+
+def get_urgency_if_urgent(pr) -> Optional[ReviewDelta]:
+    updated_at = datetime.fromisoformat(pr["updated_at"].rstrip("Z"))
+    today = datetime.now()
+    urgency = pr_urgency(pr)
+    days = days_without_weekends(today, today - updated_at)
+
+    return ReviewDelta(urgency, days) if days > urgency.days else None
+
+
+def has_already_reviewed(request, reviews):
+    return request["login"] in [review["user"]["login"] for review in reviews]
+
+
+COMMENT_MARKER = (
+    "This reminder is being automatically generated due to the urgency configuration."
+)
+
+
+COMMENT_TEMPLATE = (
+    """
+Based on the {urgency_label} urgency of this PR, the following reviewers are being
+gently reminded to review this PR:
+
+{user_logins}
+
+"""
+    f"{COMMENT_MARKER}"
+    """
+Ignoring weekend days, this PR was updated {days_since_update} day(s) ago. PRs
+labelled with {urgency_label} urgency are expected to be reviewed within {urgency_days}.
+
+@{pr_author}, if this PR is not ready for a review, please draft it to prevent reviewers
+from getting further unnecessary pings.
+"""
+)
+
+
+def build_comment(review_delta: ReviewDelta, stale_requests, pr):
+    user_handles = [f"@{req['login']}" for req in stale_requests]
+    return user_handles, COMMENT_TEMPLATE.format(
+        urgency_label=review_delta.urgency.label,
+        urgency_days=review_delta.urgency.days,
+        user_logins="\n".join(user_handles),
+        days_since_update=review_delta.days,
+        pr_author=pr["user"]["login"],
+    )
+
+
+def base_repo_name(pr):
+    return pr["base"]["repo"]["name"]
+
+
+def post_reminders(github_pat: str, dry_run: bool):
+    if datetime.now().weekday() >= 5:
+        return  # it's the weekend!
```

Comment:
We could potentially set this DAG's schedule interval to [every weekday](https://crontab.guru/every-weekday), that would prevent us from having empty runs on the weekend! If we want to keep it at `@daily` though, I'd suggest raising an [`AirflowSkipException`](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/exceptions/index.html#airflow.exceptions.AirflowSkipException) - this will make it easier to tell from the DAG UI which runs didn't actually process any data.